pkg/api/types.go

```diff
@@ -983,6 +994,16 @@ type Binding struct {
 	Target ObjectReference `json:"target"`
 }
 
+// DeleteOptions may be provided when deleting an API object
+type DeleteOptions struct {
+	TypeMeta `json:",inline"`
+
+	// Optional duration in seconds before the object should be deleted. Value must be non-negative integer.
+	// The value zero indicates delete immediately. If this value is nil, the default grace period for the
+	// specified type will be used.
+	GracePeriodSeconds *int64 `json:"gracePeriodSeconds"`
```

Comment:
Correct.  All the other constructs were more awkward (struct wrapped type, etc).


> On Mar 18, 2015, at 8:38 PM, Brian Grant <notifications@github.com> wrote:
> 
> In pkg/api/types.go:
> 
> > @@ -983,6 +994,16 @@ type Binding struct {
> >  	Target ObjectReference `json:"target"`
> >  }
> >  
> > +// DeleteOptions may be provided when deleting an API object
> > +type DeleteOptions struct {
> > +	TypeMeta `json:",inline"`
> > +
> > +	// Optional duration in seconds before the object should be deleted. Value must be non-negative integer.
> > +	// The value zero indicates delete immediately. If this value is nil, the default grace period for the
> > +	// specified type will be used.
> > +	GracePeriodSeconds *int64 `json:"gracePeriodSeconds"`
> This is a pointer so we can distinguish Go defaulting from the user specifying 0?
> 
> —
> Reply to this email directly or view it on GitHub.
>